src/mpid/ch4/shm/src/shm_control.h

```diff
@@ -13,11 +13,17 @@
 #define SHM_CONTROL_H_INCLUDED
 
 #include "shm_types.h"
+#include "../xpmem/xpmem_pre.h"
 #include "../posix/posix_am.h"
 
 #ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
 MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_ctrl_send_lmt_ack_cb(MPIDI_SHM_ctrl_hdr_t * ctrl_hdr);
 MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_ctrl_send_lmt_req_cb(MPIDI_SHM_ctrl_hdr_t * ctrl_hdr);
+#ifdef MPIDI_CH4_SHM_XPMEM_COOP_P2P
```

Comment:
I'm not sure the additional `#ifdef`s are necessary. I think the better route is to have a CVAR to turn this optimization on or off. Either way, I think it should be compiled in to keep the code simpler. Can you take the `#ifdef MPIDI_CH4_SHM_XPMEM_COOP_P2P` out of all of the files?

Reply:
`MPIDI_CH4_SHM_XPMEM_COOP_P2P` also controls whether we need to compile in the extra data field specific for COOP P2P in a `MPIDI_XPMEM_global_t` and `MPIDI_XPMEM_am_request_t`, it is better to use this macro or we may waste memory when we don't use COOP P2P.

Reply:
get it.